packages/Table/src/TableHead/SortableCell/SortableCell.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useState } from "react";
+
+const SortableCell = ({
+  onSort,
+  orderBy,
+  column,
+  columns
+}: {
+  onSort: any;
+  orderBy: any;
+  column: any;
+  columns: any;
+}) => {
+  const formatColumns = columns.reduce((acc: any, item: any): any => {
+    return { ...acc, [item]: null };
+  }, {});
+  const [tableColumns, setTableColumns] = useState(formatColumns);
```

Comment:
I think hooks need to be placed in the first lines of the function (here it would be line 14)

Reply:
If I do that probably I would introduce hoisting here that is works to be compliant with this rule.